clean_air/util/file_converter.py

```diff
@@ -0,0 +1,106 @@
+"""This converts xcel data files to JSON file types for processing."""
+import pandas as pd
+import openpyxl
+import json
+import datetime
+import xarray as xr
+from json import JSONEncoder
+from collections import OrderedDict
+
+
+def read_excel_data(filepath):
+    """
+    Reads in data from excel spreadsheets and holds as temporary
+    pandas.DataFrame object.
+    """
+    temp_dataframe = pd.read_excel(filepath, engine='openpyxl')
+    print("reading today's excel data file and committing to memory...")
+    return pd.DataFrame(temp_dataframe)
+
+
+def read_netcdf_data(filepath):
+    """Reads in netcdf data from somewhere and holds in an xarray
+    something..."""
+    # TODO: Find details of ticket and complete the description of this function
+    temp_dataframe = xr.open_dataset(filepath)
+    print("reading today's netcdf data file and committing to memory...")
+    return xr.DataArray(temp_dataframe)
+
+
+def save_as_json(self):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as JSON string.
+    """
+    # iterate through each row to split into separate dataframes and save in
+    # clean_air/data/json_data/:
+    for r, row in enumerate(temp_xl_df.iterrows()):
+        print('extracting data from row ', r, ' right now...')
+
+        # obtain data from excel file:
+        self.title = row[1].values[16]  # title of the model, not the user.
+        self.description = row[1].values[17]  # again, description of model.
+        # self.cs = ?
+        self.firstname1 = row[1].values[6]
+        self.surname1 = row[1].Surname
+        self.firstname2 = row[1].values[11]
+        self.surname2 = row[1].values[12]
+        self.north = row[1].values[42]
+        self.south = row[1].values[41]
+        self.east = row[1].values[43]
+        self.west = row[1].values[44]
+        self.chemicals = row[1].values[46]
+        self.obs_level = row[1].values[19]
+        self.data_source = row[1].values[20]
+        self.time_range_start = row[1].values[37]
+        self.time_range_end = row[1].values[38]
+        self.lineage = row[1].values[50]
+        self.quality = row[1].values[51]
+        self.docs = row[1].values[22]
+
+        # turn this into an OrderedDict first, then save to json:
+        new_file = OrderedDict({"title": self.title,
+                                "description": self.description,
+                                "authors":
+                                    ["firstname", self.firstname1,
+                                     "surname", self.surname1,
+                                     "firstname2", self.firstname2,
+                                     "surname2", self.surname2],
```

Comment:
Getting the formatting right for this was really difficult.  The yaml that Elle made only had space for two named contributors, and I think the metadata form itself has the same setup: https://metoffice.sharepoint.com/sites/scienceweatheradaqteam/Projects/CleanAir/WP2b_WP4_work/framework/Metadata/Clean%20Air%20Framework%20-%20Metadata%20Form.pdf?CT=1628501597739&OR=ItemsView

So I understand that you are suggesting that I change the output of the name section to multiple separate dictionaries (I might be able to do that), what exactly are you suggesting if the unused fields are left blank?  I think I can tell the JSON handler to exclude nans if that's better?

Reply:
Still having issues trying to get a list of dicts in the output file.  If i use dicts instead of sets inside the main dictionary then they get interpreted as 'null'.  What exactly is the significance of these variables? i.e. how are they going to be used and what difference will it make if they are saved as sets not dicts?

Reply:
Don't bother responding to that, I got it sorted anyway.  All lists of info are now in dicts.